homeassistant/components/ws66i/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""The Soundavo WS66i 6-Zone Amplifier integration."""
+import logging
+
+from pyws66i import get_ws66i
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .config_flow import validate_input
+from .const import CONF_NOT_FIRST_RUN, DOMAIN, FIRST_RUN, WS66I_OBJECT
+
+PLATFORMS = ["media_player"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Soundavo WS66i 6-Zone Amplifier from a config entry."""
+
+    # double negative to handle absence of value
+    first_run = not bool(entry.data.get(CONF_NOT_FIRST_RUN))
+    if first_run:
+        hass.config_entries.async_update_entry(
+            entry, data={**entry.data, CONF_NOT_FIRST_RUN: True}
+        )
+    else:
+        # Need to verify we can connect
+        try:
+            _ = await validate_input(hass, entry.data)
+        except ConnectionError as err:
+            raise ConfigEntryNotReady from err
```

Comment:
It looks like you are creating the object twice.  Once here and once below.

We should only create it once.

`validate_input` is purpose built for the config flow.  If you want to reuse it, break it apart into multiple functions so it only does what you need here.